cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -1275,11 +1275,10 @@ impl MachInstEmit for Inst {
                     amo: AMO::SeqCst,
                 }
                 .emit(&[], sink, emit_info, state);
-                let origin_value = if ty.bits() < 32 {
-                    AtomicOP::extract(t0, offset, dst.to_reg(), ty)
+                if ty.bits() < 32 {
+                    AtomicOP::extract(dst, offset, dst.to_reg(), ty)
                         .iter()
                         .for_each(|i| i.emit(&[], sink, emit_info, state));
-                    t0.to_reg()
                 } else if ty.bits() == 32 {
                     Inst::Extend {
                         rd: t0,
```

Comment:
Isn't this case now wrong since we are extending into `t0` instead of `dst`?

Reply:
Yes.